### PR TITLE
Garden EnvMeasurements Index

### LINKED ISSUE #1: Create Gardens Table in Database
Create Gardens table with latitude, longitude and name columns

### LINKED ISSUE #2: Create env_measurements Table
Create Env_points table with soil_temp and soil_moisture. This table belongs to gardens.

### LINKED ISSUE #7: Deploy BE App
- [ ]  Deployed to Heroku

### LINKED ISSUE #8: Implement TravisCI
- [ ] Complete

### LINKED ISSUE #9: Garden EnvMeasurements Index Endpoint
User can see all env_measurements associated with a garden by sending garden id to `GET /api/v1/gardens/:id/env_measurements`. The response returns all env measurements associated with the garden

Response format:
```
{
    "data": [
        {
            "id": "1",
            "type": "env_measurement",
            "attributes": {
                "soil_temperature": 55.52,
                "soil_moisture": 85.33,
                "created_at": "2019-09-01T19:39:52.192Z"
            }
        },
        {
            "id": "2",
            "type": "env_measurement",
            "attributes": {
                "soil_temperature": 57.28,
                "soil_moisture": 83.22,
                "created_at": "2019-09-01T19:54:52.204Z"
            }
        },
        {
            "id": "3",
            "type": "env_measurement",
            "attributes": {
                "soil_temperature": 59.11,
                "soil_moisture": 81.11,
                "created_at": "2019-09-01T20:09:52.206Z"
            }
        }
    ]
}
```
### Checks
- [x] For success, returns status code 200
- [x] Returns correct response format json

Sad path
- [x] For error, returns status code 404 if garden is not found

### PR DESCRIPTION
**Pull request by Mycobee**
## Changes proposed in this pull request:
* This PR adds `GET /api/v1/gardens/:id/env_measurments` happy path and sad path functionality
* This PR adds garden and env_measurments models and specs, as well as a seedfile
* This PR adds testing setup, and travisCI configuration

## Card(s) closed in this pull request:
close #9 
close #7 
close #1 
close #2 
close #8 

## What did you struggle on to complete?
* Deciding how to the structure the API was a bit challenging.  Initially we wanted one endpoint to handle both gardens and env_measurements, but we went back and restructured it in a way that is more RESTful, where every resource has an individual endpoint.  

### Test Coverage Percentage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

## Helpful Resources:
[TravisCi documentation](https://docs.travis-ci.com/user/languages/ruby/#What-This-Guide-Covers)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,10 +27,17 @@ gem 'bootsnap', '>= 1.1.0', require: false
 
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
 # gem 'rack-cors'
+gem 'active_model_serializers'
```
Comment (bplantico) [inline Gemfile]: @Mycobee I'm not familiar with this gem 'active_model_serializers'.
What are the pros/cons compared to something like fast_jsonapi?

File: app/controllers/api/v1/gardens/env_measurements_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::Gardens::EnvMeasurementsController < ApplicationController
+	def index
+		garden = Garden.find_by(id: params[:id])
```
Comment (bplantico) [inline app/controllers/api/v1/gardens/env_measurements_controller.rb]: @Mycobee Since it's displaying a single garden and not multiple gardens, should this endpoint be a `show` instead of an `index` ?

Then it could be `garden = Garden.find(params[:id])`

File: app/controllers/api/v1/gardens/env_measurements_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+		not_found if garden.nil?
```
Comment (bplantico) [inline app/controllers/api/v1/gardens/env_measurements_controller.rb]: So clean!

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+	namespace :api do
+    namespace :v1 do
+			namespace :gardens do
+				get '/:id/env_measurements', to: 'env_measurements#index'
```
Comment (bplantico) [inline config/routes.rb]: Ahh now I see why this got called an index. I'm wondering if this should be backed out of `api/v1/gardens/:id/env_measurements` as an `env_measurements` index and instead route to `api/v1/gardens/:id` as a `gardens` show?

I'm thinking `/gardens/:id` feels more RESTful and inherently one garden shows all of it's measurements.
Also wondering if we did keep `/gardens/:id/env_,easurements`, would we ever take it one more step to show a single env_measurement at a show `/gardens/:id/env_measurements/:id`?

File: app/controllers/api/v1/gardens/env_measurements_controller.rb (hunk 2 above)
Comment (Mycobee) [reply app/controllers/api/v1/gardens/env_measurements_controller.rb]: The reason I went this route, when an ID is not found with find, it throws an error immediately.  Using find_by allows us to do more with a nil object for custom error handling.  I am currently working on taking this logic out of the controller to keep them skinny :)

File: app/controllers/api/v1/gardens/env_measurements_controller.rb (hunk 3 above)
Comment (Mycobee) [reply app/controllers/api/v1/gardens/env_measurements_controller.rb]: thanks.  I love shorthand conditionals

File: config/routes.rb (hunk 4 above)
Comment (Mycobee) [reply config/routes.rb]: Good point.  I think breaking them out in to separate endpoints is a better long term strategy.  I am working on revising it now.  Thanks for your input

File: app/controllers/api/v1/gardens/env_measurements_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::Gardens::EnvMeasurementsController < ApplicationController
+  before_action :set_garden, only: [:index]
+
+	def index
+    render json: EnvMeasurementSerializer.new(@garden.env_measurements)	
+	end	
+
+  private
+
+  def set_garden
```
Comment (bplantico) [inline app/controllers/api/v1/gardens/env_measurements_controller.rb]: Nice move here. Really like the abstraction.

File: app/controllers/api/v1/gardens/env_measurements_controller.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class Api::V1::Gardens::EnvMeasurementsController < ApplicationController
+  before_action :set_garden, only: [:index]
+
+	def index
+    render json: EnvMeasurementSerializer.new(@garden.env_measurements)
```
Comment (bplantico) [inline app/controllers/api/v1/gardens/env_measurements_controller.rb]: Changing this to EnvMeasurementSerialize definitely makes this clearer than sending it to a GardenSerializer, awesome work Rob.

Thanks for talking through this change with me. I definitely see you were right about nesting/namespacing to env_measurements how you did.

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::API
+	def not_found
+    render json: { error: "Garden Not Found" }, status: :not_found
```
Comment (bplantico) [inline app/controllers/application_controller.rb]: Brilliant.

File: config/routes.rb (hunk 4 above)
Comment (bplantico) [reply config/routes.rb]: I appreciate being able to talk through this together @Mycobee! Thanks for explaining your reasoning, I definitely see your point about namespacing this to an index the way you did instead of returning env_measurements back as part of a garden.

Continuing to think more about this, if we did change this we could make this a similar call to DarkSky's forecast endpoint where it's possible to limit the number of hour/day objects returned for a forecast, but that definitely increases the complexity so for now I like the path we're on.
